internal/controlplane/handlers_user.go

```diff
@@ -272,53 +273,41 @@ func (s *Server) ListInvitations(ctx context.Context, _ *pb.ListInvitationsReque
 	invitations := make([]*pb.Invitation, 0)
 
 	// Extracts the user email from the token
-	tokenString, err := gauth.AuthFromMD(ctx, "bearer")
-	if err != nil {
-		return nil, status.Errorf(codes.InvalidArgument, "no auth token: %v", err)
-	}
-	token, err := s.jwt.ParseAndValidate(tokenString)
+	tokenEmail, err := auth.GetUserEmailFromContext(ctx)
 	if err != nil {
-		return nil, status.Errorf(codes.Unauthenticated, "failed to parse bearer token: %v", err)
+		return nil, status.Errorf(codes.Internal, "failed to get user email: %s", err)
 	}
 
 	// Get the list of invitations for the user
-	invites, err := s.store.GetInvitationsByEmail(ctx, token.Email())
+	invites, err := s.store.GetInvitationsByEmail(ctx, tokenEmail)
```

Comment:
This works only for invites for the address the user has in their token right?

Reply:
Is there a way to list the invites by the identity instead and look up the identity by e-mail?

Reply:
Should we expect that the same user identity can have multiple emails in their token (they get that from Keycloak right?) The idea was to check that for listing invitations, but still allow the user to accept/decline invitations by their code (without cross-checking the email)

Reply:
OK, maybe I'm missing what is the list invitations for. Is it supposed to be used by the user after they had logged in? Or by the UI?

I'm concerned about 2 things with respect to the invitations - since we would get the e-mail from keycloak we 1) can't rely on having the e-mail available I think or 2) it could be unrelated to the e-mail where invites are supposed to be sent (e.g. my e-mail address is available but it's my personal e-mail)

Reply:
OK, but looking at the invite table, I don't think we have any other data to cross-reference the invite with. So this is more to be careful where/how we use that endpoint - it might not give us a complete list I think.

Reply:
Yes, this is supposed to be called from a logged in user to check their pending invitations (both cli and ui). My assumption is we can rely on having the mail in the token since I don't think we have a use case where this is not true, i.e logging through GitHub does set the primary mail there. If the invite is sent to your company email and for whatever reason you want to accept it from your minder account using your personal mail you can still do that through the accept/decline flow but you are correct that it will not be shown via ListInvitations (so far this is the intended behaviour).

Reply:
Also invitations work solely through using the email(for the time being you cannot send an invite to a user using their minder account)